### DISCUSSION TITLE
Expressão inválida: +

### DISCUSSION THREAD

**Discussion by MathewsAlencar** [dgadelha/Portugol-Webstudio#308] [Dúvidas / Perguntas]
Desenvolvi o código abaixo, porém o portugol está me informando que a expressão no código está invalida.

`// Função para gerar números aleatórios
    funcao inteiro gerarAleatorio(inteiro min, inteiro max)
    {
        real valorAleatorio = aleatorio() // Gera um valor real entre 0 e 1
        inteiro resultado = min + inteiro(valorAleatorio * (max - min + 1)) // Ajusta para o intervalo
        retorne resultado
    }`

Já tentei várias outras alternativas, porém sem sucesso.

`programa
{
    // Definição do labirinto
    inteiro labirinto[5][5]
    inteiro posX = 0
    inteiro posY = 0
    inteiro tesouroX
    inteiro tesouroY

    // Função principal
    funcao inicio()
    {
        // Inicializa o labirinto
        inicializarLabirinto()

        // Define a posição do tesouro
        definirTesouro()

        // Loop para mover o explorador até encontrar o tesouro
        enquanto (nao encontrouTesouro())
        {
            mostrarLabirinto()
            moverExplorador()
        }

        escreva("Parabéns! Você encontrou o tesouro!\n")
    }

    // Função para inicializar o labirinto
    funcao inicializarLabirinto()
    {
        // Inicializa o labirinto com caminhos livres (0) e paredes (1)
        para (inteiro i = 0; i < 5; i++)
        {
            para (inteiro j = 0; j < 5; j++)
            {
                labirinto[i][j] = 0 // Caminho livre
            }
        }

        // Define algumas paredes
        labirinto[2][2] = 1 // Parede
        labirinto[1][1] = 1 // Outra parede
    }

    // Função para definir a posição do tesouro aleatoriamente
    funcao definirTesouro()
    {
        // Gera posições aleatórias para o tesouro
        enquanto (verdadeiro)
        {
            tesouroX = gerarAleatorio(0, 4) // Gera a posição X do tesouro
            tesouroY = gerarAleatorio(0, 4) // Gera a posição Y do tesouro
            se (labirinto[tesouroX][tesouroY] == 0) // Verifica se a posição é livre
            {
                labirinto[tesouroX][tesouroY] = 2 // Coloca o tesouro na posição
                pare
            }
        }
    }

    // Função para verificar se o explorador encontrou o tesouro
    funcao logico encontrouTesouro()
    {
        retorne (posX == tesouroX e posY == tesouroY)
    }

    // Função para mover o explorador
    funcao moverExplorador()
    {
        caractere movimento
        escreva("Para onde deseja se mover? (c: cima, b: baixo, e: esquerda, d: direita): ")
        leia(movimento)

        // Atualiza a posição com base na entrada do usuário
        se (movimento == 'c' e posX > 0) // Cima
        {
            se (labirinto[posX - 1][posY] != 1) // Verifica se não é parede
            {
                posX--
            }
            senao
            {
                escreva("Movimento inválido. Há uma parede na direção escolhida.\n")
            }
        }
        senao se (movimento == 'b' e posX < 4) // Baixo
        {
            se (labirinto[posX + 1][posY] != 1)
            {
                posX++
            }
            senao
            {
                escreva("Movimento inválido. Há uma parede na direção escolhida.\n")
            }
        }
        senao se (movimento == 'e' e posY > 0) // Esquerda
        {
            se (labirinto[posX][posY - 1] != 1)
            {
                posY--
            }
            senao
            {
                escreva("Movimento inválido. Há uma parede na direção escolhida.\n")
            }
        }
        senao se (movimento == 'd' e posY < 4) // Direita
        {
            se (labirinto[posX][posY + 1] != 1)
            {
                posY++
            }
            senao
            {
                escreva("Movimento inválido. Há uma parede na direção escolhida.\n")
            }
        }
        senao
        {
            escreva("Movimento inválido. Tente novamente.\n")
        }

        escreva("Posição atual: [", posX, "][", posY, "]\n")
    }

    // Função para mostrar o labirinto
    funcao mostrarLabirinto()
    {
        para (inteiro i = 0; i < 5; i++)
        {
            para (inteiro j = 0; j < 5; j++)
            {
                se (i == posX e j == posY)
                {
                    escreva("E ") // E representa o explorador
                }
                senao se (labirinto[i][j] == 1)
                {
                    escreva("# ") // # representa uma parede
                }
                senao se (labirinto[i][j] == 2)
                {
                    escreva("T ") // T representa o tesouro
                }
                senao
                {
                    escreva(". ") // . representa um caminho livre
                }
            }
            escreval() // Nova linha após cada linha do labirinto
        }
    }

    // Função para gerar números aleatórios
    funcao inteiro gerarAleatorio(inteiro min, inteiro max)
    {
        real valorAleatorio = aleatorio() // Gera um valor real entre 0 e 1
        inteiro resultado = min + inteiro(valorAleatorio * (max - min + 1)) // Ajusta para o intervalo
        retorne resultado
    }
}
`

**Comment by WilCb** [comment]
Onde está a função `aleatorio()` ? Você está chamando ela, mas não foi criada
